### PR TITLE
Timetable revamp

### PR DESCRIPTION
**Pull request by yangshun**
- Rewrite in (almost) non-flex approach. 
- Add a toggle orientation button.

### CODE REVIEW

File: v3/src/js/reducers/theme.js [hunk 1]
```diff
@@ -2,16 +2,26 @@
...
+/* eslint-disable no-duplicate-imports */
+import type {
+  ColorIndex,
+  ColorMapping,
+  ThemeState,
+} from 'types/reducers';
+import {
+  VERTICAL,
+  HORIZONTAL,
+} from 'types/reducers';
```
Comment (li-kai) [inline v3/src/js/reducers/theme.js]: Just merge with the above? Why import twice? The constants should be ranked higher up too in any case.

File: v3/src/js/views/timetable/Timetable.jsx [hunk 2]
```diff
@@ -1,43 +1,113 @@
 // @flow
 
+import type { DayText, LessonTime, RawLesson } from 'types/modules';
+import type { TimetableArrangement } from 'types/timetables';
+
 import React, { Component } from 'react';
```
Comment (li-kai) [inline v3/src/js/views/timetable/Timetable.jsx]: React and module dependencies should be imported first, then followed by types.

File: v3/src/js/views/timetable/TimetableContainer.jsx [hunk 3]
```diff
@@ -1,11 +1,28 @@
 // @flow
+/* eslint-disable no-duplicate-imports */
+import type {
```
Comment (li-kai) [inline v3/src/js/views/timetable/TimetableContainer.jsx]: Types should really be closer to the code as it makes it easier to reference where the types come from.

File: v3/src/js/views/timetable/TimetableRow.jsx [hunk 4]
```diff
@@ -40,16 +39,25 @@ function generateCells(lessons?: Array<Lesson>, onModifyCell?: Function) {
 }
 
 type Props = {
-  day: string,
+  cellSize: number,
+  cellOrientationStyleProp: string,
+  horizontalOrientation?: boolean,
+  proportion?: number,
```
Comment (li-kai) [inline v3/src/js/views/timetable/TimetableRow.jsx]: Would `scale` be better? `props.proportion` sounds like it could be confusing.

Comment (yangshun) [inline v3/src/js/views/timetable/TimetableRow.jsx]: Ok `scale` it is.

**Pull request merged**